### PR TITLE
fix: Profile Sidebar keeps opened after navbar select 

### LINKED ISSUE #9255: Profile Sidebar keeps opened after navbar select
### What happened?

desktop

CleanShot 2024-02-03 at 09 48 13

mobile

CleanShot 2024-02-03 at 09 59 52

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Unified events triggered at the same time and changed them to `@select` 
- [x] Closes #9255

## Needs QA check

@prury pls  test the navbar (desktop and mobile) to see if they work fine 

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2024-02-03 at 10 18 33

### CODE REVIEW

**Comment by prury** [comment]
@hassnian still happens when i click on drops or on koda logo to go to landing

**Comment by hassnian** [comment]
@prury try again , all links from navbar and on search bar enter , thanks

File: components/Navbar.vue [hunk 1]
```diff
@@ -9,7 +9,10 @@
     <div class="container items-center" :class="{ 'is-fluid': !isTouch }">
       <!-- BRAND -->
       <div class="navbar-brand">
-        <nuxt-link to="/" class="navbar-item logo nuxt-link-active">
+        <nuxt-link
+          to="/"
+          class="navbar-item logo nuxt-link-active"
+          @click="closeAllModals">
```
Comment (roiLeo) [inline components/Navbar.vue:15]: should it be like
```suggestion
          @click.prevent="closeAllModals">
```
?

**Review by roiLeo — APPROVED** [review]
or we could do that using middleware, each navigation occur => `closeAllModal`

File: tests/e2e/sidebar.spec.ts
Comment (preschian) [inline tests/e2e/sidebar.spec.ts:1]: nice 👍🏻

**Review by preschian — APPROVED** [review]

**Comment by hassnian** [comment]
> or we could do that using middleware, each navigation occur => `closeAllModal`

thought the same, way cleaner didn't know if we had/wanted a modal that was kept open on route change.

**Comment by roiLeo** [comment]
> thought the same, a cleaner solution tbh
> 
> didn't know if we had/wanted a modal that was kept open on route change.
> 
> @roiLeo wdyt

I don't see a case where we would keep it open on page load, so might be easier to add in middleware

File: middleware/modal.global.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default defineNuxtRouteMiddleware(() => {
+  const nuxtApp = useNuxtApp()
```
Comment (roiLeo) [inline middleware/modal.global.ts:2]: better?
```suggestion
  const { isHydrating } = useNuxtApp()
```

**Review by roiLeo — APPROVED** [review]

File: middleware/modal.global.ts (hunk 2 above)
Comment (hassnian) [reply middleware/modal.global.ts]: yep ✨

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 6.88 USD/DOT ~ 7.267 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0xebe1bc31dfa8dabff4bb401ae09a75f675d592aee686dfe8790a7f326aeb5a06](https://polkadot.subscan.io/extrinsic/0xebe1bc31dfa8dabff4bb401ae09a75f675d592aee686dfe8790a7f326aeb5a06)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**